src/etc/inc/openvpn.inc

```diff
@@ -1096,8 +1106,32 @@ function openvpn_reconfigure($mode, $settings) {
 					$tls_directive = "tls-auth";
 					if ($mode == "server") {
 						$tlsopt = 0;
+						//switch($settings['tlsauth_keydir']){
+							//case '-1':
+							//case '0':
+								//$tlsopt = 0;
+								//break;
+							//case '1':
+								//$tlsopt = 1;
+								//break;
+							//case '2':
+								//$tlsopt = '';
+								//break;
+						//}
 					} else {
 						$tlsopt = 1;
+						switch($settings['tlsauth_keydir']){
+							case '-1':
+							case '1':
+								$tlsopt = 1;
+								break;
+							case '0':
+								$tlsopt = 0;
+								break;
+							case '2':
+								$tlsopt = '';
+								break;
+						}
```

Comment:
Switch needs a default case, which could also be used to apply the previous/existing default behavior.